packages/components/src/color-picker/hsl-input.tsx

```diff
@@ -3,14 +3,61 @@
  */
 import { colord } from 'colord';
 
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect } from '@wordpress/element';
+
 /**
  * Internal dependencies
  */
 import { InputWithSlider } from './input-with-slider';
 import type { HslInputProps } from './types';
 
 export const HslInput = ( { color, onChange, enableAlpha }: HslInputProps ) => {
-	const { h, s, l, a } = color.toHsl();
+	const colorPropHSL = color.toHsl();
+
+	const [ internalHSLA, setInternalHSLA ] = useState( { ...colorPropHSL } );
+
+	const isInternalColorSameAsReceivedColor = color.isEqual(
+		colord( internalHSLA )
+	);
+
+	useEffect( () => {
+		if ( ! isInternalColorSameAsReceivedColor ) {
+			// Keep internal HSLA color up to date with the received color prop
+			setInternalHSLA( colorPropHSL );
+		}
+	}, [ colorPropHSL, isInternalColorSameAsReceivedColor ] );
```

Comment:
Feels like one of those situations where the `react-hooks/exhaustive-deps` check is getting ahead of itself! `colorPropsHSL` will always be a different object, and we're checking `isInternalColorSameAsReceivedColor` in the callback anyway.

Reply:
We could use a string representation (`.toHslString()`) and that would be a more stable dependency. Then, inside the effect's condition, we could do something like `setInternalHSLA( colord( hslString ).toHsl() )`. Alternatively, we could memoize the HSL object and rebuild it only when the string representation is changed.

Reply:
Sounds good, I will memoize `colorPropHSLA` which should hopefully be the solution which reduces the amount of transformations between color spaces via the `colord` library (and therefore, hopefully the most perfoarmant)

Reply:
Done in https://github.com/WordPress/gutenberg/pull/57555/commits/d4c34dcc00c453f25dfb780c3380e7013f6677d9